packages/scan/src/core/monitor/performance-store.ts

```diff
@@ -0,0 +1,115 @@
+import { BoundedArray } from 'src/core/monitor/performance-utils';
+import {
+  InternalInteraction,
+  PerformanceInteraction,
+} from 'src/core/monitor/types';
+
+// forgot why start time was here ngl
+type Item = any;
+type UnSubscribe = () => void;
+type Callback = (item: Item) => void;
+type Updater = (state: BoundedArray<Item>) => BoundedArray<Item>;
+type ChanelName = string;
+
+type PerformanceEntryChannelsType = {
+  subscribe: (to: ChanelName, cb: Callback) => UnSubscribe;
+  publish: (
+    item: Item,
+    to: ChanelName,
+    dropFirst: boolean,
+    createIfNoChannel: boolean,
+  ) => void;
+  channels: Record<
+    ChanelName,
+    { callbacks: BoundedArray<Callback>; state: BoundedArray<Item> }
+  >;
```

Comment:
I think you can do something like this, and use generics here.

```suggestion
  channels: Record<ChannelName, Channel<T>>;
```

and you could create an interface for it like this.. 

```
interface Channel<T> {
  callbacks: BoundedArray<Callback<T>>;
  state: BoundedArray<T>;
}
```